### PR TITLE
Add colorlog and colorama dependency for windows

### PR DESCRIPTION
**Pull request by kwilcox**

### CODE REVIEW

File: recipes/colorama/bld.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+"%PYTHON%" setup.py install --single-version-externally-managed --record record.txt
+if errorlevel 1 exit 1
```
Comment (msarahan) [inline recipes/colorama/bld.bat]: you don't need bld.bat anymore (as of conda-build 1.20.0).  the build: script: in meta.yaml suffices.

File: recipes/colorlog/bld.bat [hunk 2]
```diff
@@ -0,0 +1,2 @@
+"%PYTHON%" setup.py install --single-version-externally-managed --record record.txt
+if errorlevel 1 exit 1
```
Comment (msarahan) [inline recipes/colorlog/bld.bat]: same story here: you don't need bld.bat anymore (as of conda-build 1.20.0). the build: script: in meta.yaml suffices.

File: recipes/colorlog/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+    build:
+        - python
+        - setuptools
+
+    run:
+        - python
+        - colorama [win]
```
Comment (msarahan) [inline recipes/colorlog/meta.yaml]: - colorama  # [win]

**Comment by kwilcox** [comment]
Example recipe should be updated, I'm not thinking many people will be following the internals of conda-build.

File: recipes/colorama/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{% set name = "colorama" %}
+{% set version = "0.3.7" %}
+{% set build = 0 %}
+{% set md5 = "349d2b02618d3d39e5c6aede36fe3c1a" %}
```
Comment (msarahan) [inline recipes/colorama/meta.yaml]: I prefer sha256, but this is a minor thing.  md5 is broken from a security standpoint.

Not a sticking point, just a comment - change if you want, or don't if you don't.

**Comment by msarahan** [comment]
RE: example recipe, yes.  I'll see about changing that.  Sorry if it caused you wasted effort =(

**Comment by msarahan** [comment]
LGTM.  Merging.  Thanks @kwilcox

**Comment by msarahan** [comment]
Sorry, spoke too soon.  Tests are good when they pass.

Here, you're importing ```tests```, but you're probably not actually installing it.  That's common with test suites.  You can either install it (modify setup.py), or copy it to the recipe folder, and write some test script to call it.

**Comment by kwilcox** [comment]
I can import `colorlog.tests` when installing via `pip... is your comment still valid?

**Comment by kwilcox** [comment]
ah... its a Python 2.7 thing. I can import `colorlog.tests` on `py35` but not `py27`

**Comment by msarahan** [comment]
There's probably some python version-specific stuff somewhere.  With Python 3, the tests is copied OK, but not py2.7

**Comment by msarahan** [comment]
@jakirkham @pelson @ocefpaf do you have patterns for running test suites that ship with packages and may not be packaged with said packages?

Do you copy files into the recipe?

The issue is that at test-time, you no longer have the source tree available.  So, to run tests, you might need to run them in the build script instead.

**Comment by jakirkham** [comment]
Sorry I may not be totally responsive today and tomorrow as I am working on some internal stuff today and tomorrow just as FYI and will then be out on holiday until Tuesday. Also, @ocefpaf is on holiday and won't be back until next Monday.

> Do you copy files into the recipe?

Generally, I have tried to push for recipes to stay slim. In other words, strictly what is necessary for the build (e.g. patches, `CMakeLists.txt` if needed, possibly a `Makefile`). If it gets too hairy, I'm concerned that we are increasing the maintenance burden on ourselves.

In some cases, I have needed more stuff to run tests that I need at build time and would rather not include in the recipe. So, I have added a clone step in the build itself. Certainly something similar could be tried to download or clone testing infrastructure here.

In some cases, others have needed data for an application's use and this has been broken out into a separate package and added as a dependency (e.g. Basemap data https://github.com/conda-forge/staged-recipes/pull/203 ).

Hope this helps.

**Comment by ocefpaf** [comment]
Thanks @msarahan for the review and @kwilcox for the PR.

Running further tests are on a case-by-case basis. I believe that the main module import here is fine. Feel free to add more in the feedstock if you feel the need to @kwilcox.

**Pull request merged**